### DISCUSSION TITLE
fill and select

### DISCUSSION THREAD

**Discussion by mshoek** [keiwando/toonsquid#383] [Ideas]
**a. Fill bucket - erase function**
Could be useful for erasing the background for example, and making a quick silhouette of a character. 

**b. Expand fill by x pixels** 
DinoAwakening and jsllllc suggested this earlier, it could save so much time!! 

**c. Magic wand -  also with the expand option** 

**d. Selecting by color**

**Comment by DinoAwakening** [comment]
Yeah, I know what you trying to say, but let me elaborate it a bit more:

**Rework for the Fill Bucket Tool:** 

**A) Transparent “Color” for Bucket Tool:** So we’ll be able to delete an entire area of a specific color.

**B) Expand fill by X amount of pixels:** In order to easily color an area even with texturized brushes, or just if “Tolerations” can’t handle it.

**C) Blur Filling by X amount of pixels:** I think it’s self-explanatory, It refers to fill with smooth fade at the end of the area, so filling won’t act so rough. It’s nice combined with “Expand” feature.

**Selection Improvements:**

**A) Implement a magic wand:** Being able to select an specific color, not too influenced by color tolerance system that bucket fill has.

**B) Implement photoshop-like straight lines area selection:** I don’t know if you know what am I talking about-… Is like a lasso selection for people with shaky hands XD

**C) Expand artificially your selected area by X amount of pixels:** Once again. Self-explanatory XD

Also: I attach my concepts video once again (Even thought Keiwan already saw it) https://youtu.be/i2ZlolZOH98

**Reply by mshoek** [reply]
Thanks, that's what I meant. :D 
<br />

> A) Implement a magic wand: Being able to select an specific color, not too influenced by color tolerance system that bucket fill has.

And it should have a contiguous / not contiguous option!
<br />

> B) Implement photoshop-like straight lines area 

I don't know how could I forget about this. :D
<br />
Also, cool concepts vid!
